### PR TITLE
[Pair] Present request when reparing

### PR DESCRIPTION
**Pull request by alexander-lsvk**
https://github.com/WalletConnect/WalletConnectSwiftV2/assets/15164826/280dff59-3f67-4a88-adb9-4ce646540add

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

**Review by llbartekll — APPROVED** [review]
LGTM

**Review by radeknovis — APPROVED** [review]

File: Sources/WalletConnectPairing/Services/Wallet/WalletPairService.swift [hunk 1]
```diff
@@ -9,18 +9,23 @@ actor WalletPairService {
...
         self.kms = kms
         self.pairingStorage = pairingStorage
+        self.history = history
     }
 
     func pair(_ uri: WalletConnectURI) async throws {
-        guard !hasPairing(for: uri.topic) else {
-            throw Errors.pairingAlreadyExist(topic: uri.topic)
+        guard try !pairingHasPendingRequest(for: uri.topic) else {
+            return
```
Comment (flypaper0) [inline Sources/WalletConnectPairing/Services/Wallet/WalletPairService.swift:28]: why it's not throws anymore?

Comment (flypaper0) [reply Sources/WalletConnectPairing/Services/Wallet/WalletPairService.swift:28]: with throwing error we can easily test this check

File: Sources/WalletConnectPairing/Services/Wallet/WalletPairService.swift [hunk 2]
```diff
@@ -39,9 +44,23 @@ actor WalletPairService {
 
 // MARK: - Private functions
 extension WalletPairService {
-    func hasPairing(for topic: String) -> Bool {
-        if let pairing = pairingStorage.getPairing(forTopic: topic) {
-            return pairing.requestReceived
+    func pairingHasPendingRequest(for topic: String) throws -> Bool {
```
Comment (flypaper0) [inline Sources/WalletConnectPairing/Services/Wallet/WalletPairService.swift:47]: do we need unit test for this method?

**Pull request merged**